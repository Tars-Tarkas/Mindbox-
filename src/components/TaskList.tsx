import React, { useContext } from "react";
import type { IToDo, ToDoContextType } from "../types/todo";
import { ToDoContext } from "../context/TodoContext";
import Task from "./Task";

export default function TaskList() {
  const { updateToDo, filter } = useContext(ToDoContext) as ToDoContextType;

  return (
    <ul
      style={{
        width: "100%",
        marginBottom: "30px",
        listStyleType: "none",
        marginLeft: 0,
        paddingLeft: 0,
      }}
    >
      {filter.length !== 0 ? (
        filter.map((todo: IToDo) => (
          <Task updateToDo={updateToDo} todo={todo} key={todo.id} />
        ))
      ) : (
        <div style={{ textAlign: "center" }}>
          <span>not found to do</span>
        </div>
      )}
    </ul>
  );
}
