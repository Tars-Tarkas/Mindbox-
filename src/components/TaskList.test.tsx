import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import TodoProvider, { ToDoContext } from "../context/TodoContext";
import TaskList from "./TaskList";
import type { IToDo } from "../types/todo";

const mockData = [
  {
    id: 1,
    text: "ReactJS",
    completed: false,
  },
  {
    id: 2,
    text: "NextJs",
    completed: true,
  },
  {
    id: 3,
    text: "TailwindCSS",
    completed: false,
  },
] as IToDo[];

describe("TaskList component", () => {
  it("TaskList render", () => {
    render(
      <TodoProvider>
        <TaskList />
      </TodoProvider>
    );
  });
  it("TaskList snapshot", () => {
    const taskList = render(
      <TodoProvider>
        <TaskList />
      </TodoProvider>
    );
    expect(taskList).toMatchSnapshot();
  });
  it("Render list mockdata", () => {
    const addTodo = jest.fn();
    const updateToDo = jest.fn();
    const filter = [{}] as IToDo[];
    const setFilter = jest.fn();
    const clearCompleted = jest.fn();
    render(
      <ToDoContext.Provider
        value={{
          todos: mockData,
          addTodo,
          updateToDo,
          filter,
          setFilter,
          clearCompleted,
        }}
      >
        <TaskList />
      </ToDoContext.Provider>
    );
    expect(screen.getByRole("list")).toBeInTheDocument();
  });
});
