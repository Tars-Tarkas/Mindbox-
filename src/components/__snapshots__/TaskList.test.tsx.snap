// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component TaskList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        style="width: 100%; margin-bottom: 30px; list-style-type: none; margin-left: 0px; padding-left: 0px;"
      >
        <div
          style="text-align: center;"
        >
          <span>
            not found to do
          </span>
        </div>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      style="width: 100%; margin-bottom: 30px; list-style-type: none; margin-left: 0px; padding-left: 0px;"
    >
      <div
        style="text-align: center;"
      >
        <span>
          not found to do
        </span>
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
